04-onion-routing.md

```diff
@@ -855,76 +855,70 @@ func NewOnionPacket(paymentPath []*btcec.PublicKey, sessionKey *btcec.PrivateKey
 }
 ```
 
-# Packet Forwarding
+# Onion Decryption
 
-This specification is limited to `version` `0` packets; the structure
-of future versions may change.
+There are two kinds of `onion_packet` we use: 
+1. `onion_routing_packet` in `update_add_htlc` for payments, which contains a `payload` TLV (see [Adding an HTLC](02-peer-protocol.md#adding-an-htlc-update_add_htlc))
+2. `onion_message_packet` on `onion_message` for messages, which contains a `onionmsg_tlv` TLV (see [Onion Messages](#onion-messages)
 
-Upon receiving a packet, a processing node compares the version byte of the
-packet with its own supported versions and aborts the connection if the packet
-specifies a version number that it doesn't support.
-For packets with supported version numbers, the processing node first parses the
-packet into its individual fields.
-
-Next, the processing node computes the shared secret using the private key
-corresponding to its own public key and the ephemeral key from the packet, as
-described in [Shared Secret](#shared-secret).
-
-The above requirements prevent any hop along the route from retrying a payment
-multiple times, in an attempt to track a payment's progress via traffic
-analysis. Note that disabling such probing could be accomplished using a log of
-previous shared secrets or HMACs, which could be forgotten once the HTLC would
-not be accepted anyway (i.e. after `outgoing_cltv_value` has passed). Such a log
-may use a probabilistic data structure, but it MUST rate-limit commitments as
```

Comment:
I think we should elevate this section into a concrete requirement during decryption/processing. If nodes don't maintain a replay log at all, then packets can be re-injected into the network (albeit with the exact same payment hash, so in theory nodes can just settle asap when they see it).

Reply:
Yes, it is now (though that was hard to read in the total diff), see:

```
  - if the onion is for a payment:
    - if `hmac` has previously been received:
      - if the preimage is known:
        - MAY immediately redeem the HTLC using the preimage.
      - otherwise:
        - MUST abort processing the packet and fail.
```